packages/react-native/src/plugin/withAndroid.ts

```diff
@@ -0,0 +1,89 @@
+// capture-es - bitdrift's ES SDK
+// Copyright Bitdrift, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+import { withPlugins } from '@expo/config-plugins';
+import { withAppBuildGradle, withSettingsGradle } from '@expo/config-plugins';
+import type { ConfigPlugin } from '@expo/config-plugins';
+import PluginProps from './config';
+
+const withBitdriftAppBuildGradle: ConfigPlugin<PluginProps | void> = (
+  config,
+  props,
+) => {
+  return withAppBuildGradle(config, (config) => {
+    if (!config.modResults.contents.includes('dl.bitdrift.io')) {
+      // TODO(snowp): Eventually we'd always install the plugin and gate the okhttp instrumentation
+      // config on the networkInstrumentation prop.
+      if (props && props.networkInstrumentation) {
+        // Add the capture-plugin at the very top of the file.
+        config.modResults.contents =
+          `plugins {
+    id 'io.bitdrift.capture-plugin' version '0.16.4'
+}
+
+` + config.modResults.contents;
+      }
+
+      // Define the bitdrift maven repository at the end. This is necessary to resolve the SDK dependency specified by the plugin.
+      config.modResults.contents += `
+
+repositories {
+    maven {
+        url 'https://dl.bitdrift.io/sdk/android-maven'
+        content {
+          includeGroup 'io.bitdrift'
+        }
+    }
+}
+`;
```

Comment:
I don't love how ugly it is to read multiline strings like this. Might be nice to add a util that allows them to be defined in an easier to read way. I can take a stab at some point.

Reply:
There are some helpers that expo provide that try to help like this https://github.com/expo/expo/blob/d8dd52485bc0f67b5e054b2d41a3ea28856f1c41/packages/%40expo/prebuild-config/src/plugins/unversioned/expo-splash-screen/withAndroidSplashMainActivity.ts#L25-L32 but it's still pretty painful

What makes it more annoying is trying to make it so that the generated output looks nice in terms of indentation etc.

Anything we can do to make this better would be nice

Reply:
It does not, but the generated config may be visible to people working on the app via AS so I also didn't want to make it look super ugly